torch/_inductor/codegen/cpp.py

```diff
@@ -1868,7 +1988,63 @@ def __init__(self, kernel_group):
         self.call_ranges = None
         self.picked_vec_isa: codecache.VecISA = codecache.pick_vec_isa()
 
+    def legalize_bf16(self, nodes):
+        def add_to_dtype(sub_graph: torch.fx.Graph):
+            for node in sub_graph.nodes:
+                _node: torch.fx.Node = node
+                if _node.target in ["load", "constant"]:
+                    assert len(_node.args) == 3
+                    ops = _node.args[0]
+                    # If the node is constant, the last arg is dtype
+                    load_dtype = (
+                        V.graph.get_dtype(_node.args[1])
+                        if _node.target == "load"
+                        else _node.args[-1]
+                    )
+
+                    if load_dtype == torch.bfloat16:
+                        with sub_graph.inserting_after(_node):
+                            to_type_node = sub_graph.call_method(
+                                "to_dtype", args=(ops, _node, torch.float)
+                            )
+                            to_type_node_args = to_type_node.args
+                            _node.replace_all_uses_with(to_type_node)
+                            to_type_node.args = to_type_node_args
```

Comment:
How do we handle the case where there is already `to_dtype` to `float` here. Will there be duplicated `to_dtype` on the graph and in the codegen?

Reply:
Yes, there will duplicate `to_dtype` on the graph. I will submit another PR to eliminate the useless `to_dtype` nodes.